src/utils/spinel_encoder.hpp

```diff
@@ -541,7 +538,7 @@ class SpinelEncoder
      */
     otError WriteDataWithLen(const uint8_t *aData, uint16_t aDataLen);
 
-#if OPENTHREAD_MTD || OPENTHREAD_FTD
+#if OPENTHREAD_MESSAGE_IN_SPINEL_BUFFER
```

Comment:
Can we consider adding a new header file `src/utils/ot-spinel-config.h` for all config options for spinel modules/library. This way we can add documentation (describing what the config is intended to do) and have a default value for each config (which can also help with "undefined marco warning").

Thoughts?